modules/shared/interpreter/src/main/scala/almond/interpreter/messagehandlers/InterpreterMessageHandlers.scala

```diff
@@ -168,6 +170,28 @@ final case class InterpreterMessageHandlers(
       inspectHandler
     )
 
+  def shutdownHandler: MessageHandler = {
+    // v5.3 spec states "The request can be sent on either the control or shell channels.".
+    def forChannel(c: Channel): MessageHandler = {
+      MessageHandler(c, Shutdown.requestType) { message =>
```

Comment:
To avoid having a `forChannel` method, I guess an overload of the `MessageHandler.apply` method could be added, accepting a `Set[Channel]` (and maybe passing the `Channel` to its handler, along the message). `MessageHandler`s aren't tied to specific channels, only the current `MessageHandler.apply` constructor is.